### PR TITLE
Contributing related changes/documentation

### PR DESCRIPTION
**Pull request by imaandrew**
- Add instructions for contributing
- Add CIs for testing working code and code styling
- Update some links

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Apart from glank's n64 toolchain, you will need to install clang-format and clang-tidy of at least version 14.
+
+Instuctions for adding the llvm apt repository for Ubuntu and Debian can be found [here](https://apt.llvm.org/).
+
+If the only clang-format or clang-tidy versions available for you are older than 14, you will have to build clang from source. You can find everything you need for this at the [LLVM GitHub repository](https://github.com/llvm/llvm-project)
+
+## Pull Requests
+
+Before you create a pull request with your changes, make sure your code actually builds and anything you changed works. Next, run the `format.sh` script to ensure that everything matches fp's code style. Once you've done that, you're ready to create a [pull request!](https://github.com/pmret/papermario/pulls) When creating a pull request, make sure that it is to the `develop` branch and that any commit and pull request names accurately describe what your changes are.
```
Comment (fig02) [inline CONTRIBUTING.md:15]: the pull request link is from decomp which im assuming you copy pasted from lmao
Imo no need to link to anything there, people know where the PR button is on github

**Review by fig02 — COMMENTED** [review]
also we can prob make format.sh invoke a specific clang version right?

File: CONTRIBUTING.md (hunk 1 above)
Comment (imaandrew) [reply CONTRIBUTING.md:15]: LMAO the one line i copied from decomp

**Comment by imaandrew** [comment]
> 
> 
> also we can prob make format.sh invoke a specific clang version right?

Yeah I'll just have to make sure it always gets installed as `clang-format-14` which I assume it would

**Comment by imaandrew** [comment]
Actually building it from source would probably name any version `clang-format` and not `clang-format-14`. Do you think adding a check for the version of `clang-format` if `clang-format-14` doesn't exist would be something I should add or don't bother because someone already building clang from source be able to figure out what's wrong?

File: README.md [hunk 2]
```diff
@@ -5,18 +5,18 @@ This is practice tool for Paper Mario 64 intended for use on N64 with an Everdri
 The latest release and installation instructions can be found on the [releases page](https://github.com/jcog/fp/releases).
 
 # Usage
-See the [user manual](https://github.com/JCog/fp/blob/master/USAGE.md) for information on how to use fp.
+See the [user manual](https://github.com/JCog/fp/blob/develop/USAGE.md) for information on how to use fp.
```
Comment (JCog) [inline README.md:8]: this should remain as master. until we have the manual on practicerom.com, we want to direct people toward the version for the latest release, which will always be master.

**Pull request merged**